src/androidTest/java/androidx/content/res/TypedArrayTest.kt

```diff
@@ -204,4 +204,23 @@ class TypedArrayTest {
             array.getTextOrThrow(R.styleable.TypedArrayTypes_text_array_absent)
         }.hasMessageThat().isEqualTo("Attribute not defined in set.")
     }
+
+    @Test fun use() {
+        val attrs = context.getAttributeSet(R.layout.typed_array)
+        val array = context.obtainStyledAttributes(attrs, R.styleable.TypedArrayTypes)
+
+        var booleanPresent = false
+        array.use {
+            booleanPresent = it.getBooleanOrThrow(R.styleable.TypedArrayTypes_boolean_present)
+        }
+        assertTrue(
+            "Boolean obtained from attrs wasn't correct. " +
+                "Either the 'use' block wasn't run, or the boolean was missing from the TypedArray.",
+            booleanPresent
+        )
+
+        assertThrows<RuntimeException> {
+            array.recycle()
+        }.hasMessageThat().isEqualTo("$array recycled twice!")
```

Comment:
I don't think it's worth relying on an exact error message (esp. since it's not part of the public API and therefore not guaranteed).

Reply:
Gotcha. Is there any way to check the value of `mRecycled` (a private field)?

Reply:
Only by reflection. The best thing to do here is just assert that a RuntimeException is thrown

Reply:
Will update.